stdlib/websub/src/main/ballerina/websub/Module.md

```diff
@@ -255,3 +255,99 @@ where the values specified via the Config API would override values specified as
+based on the value for the indicator in the request and a pre-defined mapping:
+
+- A request header 
+- The payload - the value for a particular key in the JSON payload
+- A request header and the payload (combination of the above)
+
+#### Samples for Resource Mapping
+
+**Based on a request header**
+
+Dispatching would be based on the value of the request header specified as `topicHeader`. 
+
+```ballerina
+websub:ExtensionConfig extensionConfig = {
+    topicIdentifier: websub:TOPIC_ID_HEADER,
+    topicHeader: "<HEADER_TO_CONSIDER>",
+    headerResourceMap: {
+        "issueOpened" : ("onIssueOpened", IssueOpenedEvent),
+        "issueClosed" : ("onIssueClosed", IssueClosedEvent)
+    }
+};
+```
+
+`"issueOpened" : ("onIssueOpened", IssueOpenedEvent)` here indicates that when the value of the header to consider is
+ `issueOpened` in the notification request, dispatching needs to happen to a resource named `onIssueOpened` whose 
+ first parameter is the generic `websub:Notification` record, and the second parameter is the custom 
+ `IssueOpenedEvent` record mapping the JSON payload received when an issue is created.  
+
+**Based on the payload**
+
+Dispatching would be based on the value in the request payload for one of the map keys specified in the 
+`payloadKeyResourceMap` map.
+
+```ballerina
+websub:ExtensionConfig extensionConfig = {
+    topicIdentifier: websub:TOPIC_ID_PAYLOAD_KEY,
+    payloadKeyResourceMap: {
+        "<PAYLOAD_KEY_TO_CONSIDER>" : {
+            "issueOpened" : ("onIssueOpened", IssueOpenedEvent),
+            "issueClosed" : ("onIssueClosed", IssueClosedEvent)
+        }
+    }
+};
+```
+
+`"issueOpened" : ("onIssueOpened", IssueOpenedEvent)` here indicates that when the value of the JSON payload key to 
+consider is `issueOpened`, dispatching needs to happen to a resource named `onIssueOpened` whose 
+ first parameter is the generic `websub:Notification` record, and the second parameter is the custom 
+ `IssueOpenedEvent` record mapping the JSON payload received when an issue is created.  
+
+**Based on a request header and the payload**
+ 
+Dispatching would be based on both a request header and the payload as specified in `headerAndPayloadKeyResourceMap`. 
+Additionally one may also introduce a `headerResourceMap` and/or `payloadKeyResourceMap` as additional mappings. 
+ 
+```ballerina
+websub:ExtensionConfig extensionConfig = {
+    topicIdentifier: websub:TOPIC_ID_HEADER_AND_PAYLOAD,
+    topicHeader: "<HEADER_TO_CONSIDER>",
+    headerAndPayloadKeyResourceMap: {
+        "issue" : {
+            "<PAYLOAD_KEY_TO_CONSIDER>" : {
+                "opened" : ("onIssueOpened", IssueOpenedEvent),
+                "closed" : ("onIssueClosed", IssueClosedEvent)
+            }
+        }
+    }
+};
+```
+`"opened" : ("onIssueOpened", IssueOpenedEvent)` here indicates that when the value of the header to consider 
```

Comment:
I found this sentence really hard to follow. Perhaps it's just too long so can we consider shortening it to make it simpler?